src/engines/duckduckgo.rs

```diff
@@ -54,16 +57,23 @@ pub async fn results(
     header_map.insert(COOKIE, "kl=wt-wt".parse()?);
 
     // fetch the html from upstream duckduckgo engine
-    // TODO: Write better error handling code to handle no results case.
     let results: String = reqwest::Client::new()
         .get(url)
+        .timeout(Duration::from_secs(30))
         .headers(header_map) // add spoofed headers to emulate human behaviour
         .send()
         .await?
         .text()
         .await?;
 
     let document: Html = Html::parse_document(&results);
+
+    let no_result: Selector = Selector::parse(".no-results")?;
+
+    if let Some(_) = document.select(&no_result).next() {
+        return Err(EngineErrorKind::EmptyResultSet);
```

Comment:
We'd better fix these warnings reported by Clippy

Reply:
Now I feel we will need to add some pre-commit checks to test these kinds of things before committing and pushing because you know this is tedious work as well as you tend to forget. also fixing these after you made the PR causes the PR to have I feel unnecessary commits also if we have a pre-commit check I think we can discuss more important stuff as well instead of formatting issues, etc. I hope you got my point :slightly_smiling_face: , So I think we need to add a pre-commit hook to check this kind of problem before making the commit. What do you say @XFFXFF ??